kartothek/serialization/_parquet.py

```diff
@@ -175,6 +189,52 @@ def restore_dataframe(
         else:
             return df
 
+    @classmethod
+    def restore_dataframe(
+        cls,
+        store: KeyValueStore,
+        key: str,
+        filter_query: Optional[str] = None,
+        columns: Optional[Iterable[str]] = None,
+        predicate_pushdown_to_io: bool = True,
+        categories: Optional[Iterable[str]] = None,
+        predicates: Optional[PredicatesType] = None,
+        date_as_object: bool = False,
+    ) -> pd.DataFrame:
```

Comment:
There should be some explanation here as to why there is a re-try, especially why there is a re-try on AssertionErrors, which is typically considered not a best practice and dangereous: AssertionErrors usually point to logic errors and logic errors can lead to invalid state of the program which is present even for the re-try. Therefore, a usual argument is to simply abort on assertion errors. So it;'s worth explaining why deviating from this here.

Reply:
I will just change the underlying error to IOError since that's more appropriate than AssertionError

Reply:
Added a comment and changed the BlockBuffer error to an error type inheriting from IOError